src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -534,6 +604,18 @@ private void extractFromFurnace(final FurnaceTileEntity furnace)
         worker.getCitizenExperienceHandler().addExperience(BASE_XP_GAIN);
     }
 
+    /**
+     * Extract fuel from the furnace.
+     *
+     * @param furnace the furnace to retrieve from.
+     */
+    private void extractFuelFromFurnace(final FurnaceTileEntity furnace)
```

Comment:
There seems to be a lot of boilerplate code for this.
Can't we just have an extractFromFurnaceSlot(furnace, slot) as one method instead of two seperate ones?

Reply:
The AbstractEntityAIUsesFurnace requires extractFromFurnace to be defined by each individual class, while the extractFuelFromFurnace is defined for all extension classes in the main one.

But I could do this in the RequestSmelter class

Reply:
Ok, I've combined the functions in the AbstractEntityAIRequestSmelter class but left it alone in the AbstractEntityAIUsesFurnace class